libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,176 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
+	solThrowIf(!isOfType<int>(_json["end"]), AssemblyImportException, "Attribute 'end' not of type int.");
+	solThrowIf(!isOfType<int>(_json["source"]), AssemblyImportException, "Attribute 'source' not of type int.");
+	solThrowIf(!isOfTypeIfExists<string>(_json, "value"), AssemblyImportException, "Optional attribute 'value' not of type string.");
+	solThrowIf(
+		!isOfTypeIfExists<int>(_json, "modifierDepth"),
+		AssemblyImportException,
+		"Optional attribute 'modifierDepth' not of type int."
+	);
+	solThrowIf(
+		!isOfTypeIfExists<string>(_json, "jumpType"),
+		AssemblyImportException,
+		"Optional attribute 'jumpType' not of type string."
+	);
+
+
+	string name = getOrDefault<string>(_json["name"], "");
+	solThrowIf(name.empty(), AssemblyImportException, "Attribute 'name' not defined or empty.");
+
+	SourceLocation location;
+	location.start = get<int>(_json["begin"]);
+	location.end = get<int>(_json["end"]);
+	int srcIndex = get<int>(_json["source"]);
+	size_t modifierDepth = static_cast<size_t>(getOrDefault<int>(_json["modifierDepth"], 0));
+	string value = getOrDefault<string>(_json["value"], "");
+	string jumpType = getOrDefault<string>(_json["jumpType"], "");
+
+	auto updateUsedTags = [&](u256 const& data)
+	{
+		m_usedTags = max(m_usedTags, static_cast<unsigned>(data) + 1);
+		return data;
+	};
+
+	auto storeImmutableHash = [&](string const& _immutableName) -> h256
+	{
+		h256 hash(util::keccak256(_immutableName));
+		solAssert(m_immutables.count(hash) == 0 || m_immutables[hash] == _immutableName);
+		m_immutables[hash] = _immutableName;
+		return hash;
+	};
+
+	auto storeLibraryHash = [&](string const& _libraryName) -> h256
+	{
+		h256 hash(util::keccak256(_libraryName));
+		solAssert(m_libraries.count(hash) == 0 || m_libraries[hash] == _libraryName);
+		m_libraries[hash] = _libraryName;
+		return hash;
+	};
+
+	if (srcIndex >= 0 && srcIndex < static_cast<int>(sourceUnitNames().size()))
+		location.sourceName = std::make_shared<std::string>(sourceUnitNames()[static_cast<size_t>(srcIndex)]);
+	else if (srcIndex < -1 || srcIndex >= static_cast<int>(sourceUnitNames().size()))
+		solThrow(AssemblyImportException, "srcIndex out of bound.");
+
+	AssemblyItem result(0);
+
+	if (c_instructions.count(name))
+	{
+		AssemblyItem item{c_instructions.at(name), location};
+		if (!jumpType.empty())
+		{
+			if (item.instruction() == Instruction::JUMP || item.instruction() == Instruction::JUMPI)
+				item.setJumpType(jumpType);
+			else
+				solThrow(
+					AssemblyImportException,
+					"Attribute 'jumpType' set on instruction different from JUMP or JUMPI (was set on instruction '" + name + "')"
+				);
+		}
+		result = item;
+	}
+	else
+	{
+		if (name == "PUSH")
+			result = {AssemblyItemType::Push, u256("0x" + value)};
+		else if (name == "PUSH [ErrorTag]")
+		{
+			solThrowIf(
+				!value.empty(),
+				AssemblyImportException,
+				"Attribute 'value' defined for instruction '" + name + "', but instruction don't need a value."
```

Comment:
```suggestion
				"Attribute 'value' defined for instruction '" + name + "', but the instruction does not need a value."
```
Same in a few more places below.